lib/task/import/csvAuthorityRecordRelationImportTask.class.php

```diff
@@ -169,45 +169,51 @@ private function import($filepath, $indexDuringImport = false, $updateMode = fal
 
   private function importRow($sourceActorId, $targetActorId, $relationTypeId)
   {
-    // Check to see if a relation of the same type already exists
-    if ($relationId = $this->getRelation($sourceActorId, $targetActorId, $relationTypeId))
+    if (!empty($this->import->status['updateMode']))
     {
-      // Ignore this data if not updating
-      if (empty($this->import->status['updateMode']))
-      {
-        return;
-      }
-
-      // Delete existing relationship, then add new, if in "delete and replace" update mode
       if ($this->import->status['updateMode'] == 'delete-and-replace')
       {
-        $relation = QubitRelation::getById($relationId);
-        $relation->delete();
+        // Handle delete-and-replace updating by deleting all existing relations between source
+        // and target then adding new relation
+        $relations = $this->getRelations($targetActorId, $sourceActorId);
+        $relationsAlternate = $this->getRelations($sourceActorId, $targetActorId);
+
+        foreach (array_unique(array_merge($relations, $relationsAlternate)) as $relationId)
+        {
+          $relation = QubitRelation::getById($relationId);
+          $relation->delete();
+        }
 
         $this->addRelation($sourceActorId, $targetActorId, $relationTypeId);
```

Comment:
Suggestion: Since the delete-and-replace, match-and-update with no match, and "no update" conditionals all result in adding a new row, you _could_ remove this call to `$this->addRelation()` and call it once after evaluating all the conditionals.  In this case, you could `return` from the match-and-update with match conditional to avoid calling `addRelation()`.  This isn't a required change imho, but would reduce the code a bit.

Reply:
Nice... yeah, that does reduce the code.